### PR TITLE
feature/11/transactions list

### PR DESCRIPTION
**Pull request by m-ossolinski**

### CODE REVIEW

File: src/components/list/TransactionItem/TransactionItem.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <p class="transaction-amount">$ 1200,00</p>
+      </div>
+      <p class="transaction-description">
+        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ut, officiis!
+      </p>
+    </StyledListItem>
+  );
+};
+
+export default TransactionItem;
```
Comment (m-ossolinski) [inline src/components/list/TransactionItem/TransactionItem.js:25]: remove export default

File: src/components/TransactionItem/TransactionItem.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { StyledListItem } from "./TransactionItem.styled";
+
+const TransactionItem = (transaction) => {
```
Comment (radekwojpl2) [inline src/components/TransactionItem/TransactionItem.js:4]: can we destruct this transaction, if you need to check how transaction looks like you have to debug, scan through code, write console log or something else, add prop types also

File: src/components/TransactionList/TransactionList.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import transactions from "./transactions.json";
+import TransactionItem from "../TransactionItem/TransactionItem";
+
+const TransactionList = () => {
+  console.log(transactions);
```
Comment (radekwojpl2) [inline src/components/TransactionList/TransactionList.js:6]: delete this, we do not need that here

File: src/components/TransactionItem/TransactionItem.js (hunk 2 above)
Comment (radekwojpl2) [reply src/components/TransactionItem/TransactionItem.js:4]: so those are props, change name to props, one naming convention is important

File: src/components/TransactionList/TransactionList.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  const renderList = transactions.map((transaction) => (
```
Comment (radekwojpl2) [inline src/components/TransactionList/TransactionList.js:8]: Create one more component, and then pass those transactions through props, then you would be able to write tests for this component and now you have strong dependency to this json file

**Review by radekwojpl2 — CHANGES_REQUESTED** [review]
enjoy

**Review by radekwojpl2 — APPROVED** [review]

File: src/components/list/TransactionItem/TransactionItem.js (hunk 1 above)
Comment (sylcym) [reply src/components/list/TransactionItem/TransactionItem.js]: done

File: src/components/list/TransactionList/TransactionList.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import TransactionItem from "../TransactionItem/TransactionItem";
+
+const TransactionList = () => {
+  return (
+    <>
+      <TransactionItem />
```
[the remark this answers is not in the document]
Comment (sylcym) [reply src/components/list/TransactionList/TransactionList.js]: done

File: src/components/TransactionList/TransactionList.js (hunk 3 above)
Comment (sylcym) [reply src/components/TransactionList/TransactionList.js]: done

File: src/components/TransactionItem/TransactionItem.js (hunk 2 above)
Comment (sylcym) [reply src/components/TransactionItem/TransactionItem.js]: done

File: src/components/list/TransactionItem/TransactionItem.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { StyledListItem } from "./TransactionItem.styled";
+
+const TransactionItem = () => {
```
[the remark this answers is not in the document]
Comment (sylcym) [reply src/components/list/TransactionItem/TransactionItem.js]: done

File: src/components/TransactionOverview/TransactionsOverview.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React, { useState } from "react";
+import transactions from "../../services/api/transactions.json";
+import { StyledOverview } from "./TransactionsOverview.styled";
+
+const getTotal = (type) => {
```
Comment (m-ossolinski) [inline src/components/TransactionOverview/TransactionsOverview.js:5]: rename it to `getTotalAmount` and create object to hold types

```
const TRANSACTIONS_TYPES = {
   expense: 'expense',
   income: 'income',
}
```

and then call it like that `getTotalAmount(TRANSACTIONS_TYPES.expense)`

File: src/components/TransactionOverview/TransactionsOverview.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+export const TransactionOverview = () => {
+  const totalExpenses = getTotal("expense");
+  const totalIncomes = getTotal("income")
+  const currentBalance = (+totalExpenses + +totalIncomes).toFixed(2);
+
+  return (
+    <StyledOverview>
+      <h3>Transactions overview:</h3>
+      <p>Total expenses: $ {totalExpenses}</p>
```
Comment (m-ossolinski) [inline src/components/TransactionOverview/TransactionsOverview.js:20]: you don't have to create separate variable to display totalExpenses 
`<p>Total expenses: $ {getTotalAmount(TRANSACTIONS_TYPES.expense)} </p>`

File: src/components/TransactionOverview/TransactionsOverview.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    .map((el) => el.amount)
+    .reduce((sum, el) => sum + el, 0)
+    .toFixed(2);
+
+  const totalIncomes = transactions
+    .filter((el) => el.type === "income")
+    .map((el) => el.amount)
+    .reduce((sum, el) => sum + el, 0)
+    .toFixed(2);
+
```
[the remark this answers is not in the document]
Comment (sylcym) [reply src/components/TransactionOverview/TransactionsOverview.js:11]: done

File: src/components/TransactionOverview/TransactionsOverview.js (hunk 7 above)
Comment (sylcym) [reply src/components/TransactionOverview/TransactionsOverview.js:5]: done

File: src/components/TransactionOverview/TransactionsOverview.js (hunk 8 above)
Comment (m-ossolinski) [reply src/components/TransactionOverview/TransactionsOverview.js]: well, I havent noticed it, you can switch back to the previous solution

Comment (sylcym) [reply src/components/TransactionOverview/TransactionsOverview.js]: ok

**Review by zakrzaq — APPROVED** [review]
:)

**Pull request merged**